mailpoet/lib/Automation/Engine/Storage/WorkflowRunStorage.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Storage;
+
+use MailPoet\Automation\Engine\Exceptions;
+use MailPoet\Automation\Engine\Workflows\WorkflowRun;
+use wpdb;
+
+class WorkflowRunStorage {
```

Comment:
I would like to have an `updateWorkflowRun($workflowRun)` method (so not sure right now, what risks there might be involved), but at least an `updateSubjects($workflowRun, $subjects)` method, so actions are able to enrich/change contextual data

Reply:
@websupporter I only added methods that the current code is using for now. I didn't try to create a "storage" service that covers all use cases, rather I added what I needed and thought we'll add more methods when we need them. Similarly for workflows — editing via the API will need to be implemented soon so we'll have to add that and think about the API.

We don't have an ORM here and so far it seemed to me these simple value objects I added served well. With editing, this may get more complicated and we'll have to decide whether the value objects should be editable via setters or an edit will mean a whole new value object and editing only particular DB columns will have specific storage methods.